app/views/job_templates/_job_template_callback_tab_headers.html.erb

```diff
@@ -0,0 +1,13 @@
+<li><a id="ansible_tab_header" href="#ansible_callback_enabled" data-toggle="tab"><%= _("Ansible") %></a></li>
+<script type="text/javascript">
+    $(document).ready(function () {
+        var provider_type = $('#job_template_provider_type');
+        provider_type.change(ansibleTabVisibilityByProvider);
+        provider_type.change();
+
+        function ansibleTabVisibilityByProvider() {
+            var tab_header = $("#ansible_tab_header");
+            $("#select2-chosen-1").text() === 'Ansible' ?  tab_header.show() : tab_header.hide();
```

Comment:
@adamruzicka this is the jquery code I added. 
Do you think we should add this line of code to the false statement (rewrite the if condition):
` $("#job_template_ansible_callback_enabled").prop('checked', false);`

This is for cases the user sets the `ansible_callback_enabled` to true and then changes the provider to `Script`. I'm not sure if it's necessary.

Reply:
That's probably not necessary. They flag might be set, but if the provider_type is not ansible, it will not be used in any way.